Add ability to post and get room queues

danoru opened the pull request.

danoru committed:
Add functionality for creating a queue with room code

danoru committed:
Add ability to create and post room queue

sburba commented on pages/api/queue/[id]/index.ts:
```diff
@@ -1,29 +1,45 @@
   } catch (e) {
     console.log(e);
-    res.status(500).json({ code: 500, message: 'ahhhh' });
+    res.status(500).json({ code: 500, message: 'Internal search error.' });
```
```suggestion
    res.status(500).json({ code: 500, message: 'Internal server error.' });
```

danoru committed:
Fixed bug with adding entries to queue

updateOne was missing an await which was stopping the database from being updated.

danoru committed:
Fixed wording in server message.

Co-authored-by: Sam Burba <github@samburba.com>
